### PR TITLE
discovery: always add chan announcements to the reject cache if err !…

### LINKED ISSUE #5191: High CPU spikes with v0.12.1-beta
### Background

We have high CPU spikes on the exact 30 minute mark of most hours throughout the day. Seems like some cron/batch process?

### Your environment

* version of `lnd`

v0.12.1-beta

* which operating system (`uname -a` on *Nix)

ubuntu 20.04

* version of `btcd`, `bitcoind`, or other backend

bitcoind (zmq) 0.21.0

* any other relevant environment details

LND is the cause of the CPU spikes verified by running `htop` during one of the spikes.
Logs don't show anything that constitutes a pattern with the spikes. 
Running an EC2 ubuntu instance t3.medium. (2 CPU, 4 GB RAM)

### Steps to reproduce
run 0.12.1-beta - using docker image

### Expected behaviour

Maybe CPU spikes every week? for auto db compaction, but not every hour.

________________
Any chance you know why this is happening?, or would you be able to instruct on how to get more information to help debug this?

### PR DESCRIPTION
**Pull request by Roasbeef**
…= ErrIgnored

In this commit, we make a change to always add chan announcements to the reject cache if we didn't reject them for already existing.  Without this change, if we end up rejecting a channel announcement say because the channel is already spent or the funding transaction doesn't exist, then we'll end up continually re-validating the same set of channels we know will fail, when they're sent to us by peers.

Fixes #5191 ?

This seems to have been an oversight in the original PR that added the cache: https://github.com/lightningnetwork/lnd/commit/bf05e4778046643d55739a755be175ed9e9402fe

### CODE REVIEW

**Comment by Roasbeef** [comment]
In addition to this, I think if we _always_ mark channels we remove due to being closed on-chain zombies, then that would tackle the problem from another angle.

**Comment by Roasbeef** [comment]
Added a unit test for this new case.

**Comment by miketwenty1** [comment]
Will let you know if this fixes #5191 within next couple days.

File: discovery/gossiper.go [hunk 1]
```diff
@@ -1741,6 +1741,10 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 			} else {
 				log.Tracef("Router rejected channel "+
 					"edge: %v", err)
+
+				d.rejectMtx.Lock()
+				d.recentRejects[msg.ShortChannelID.ToUint64()] = struct{}{}
```
Comment (wpaulino) [inline discovery/gossiper.go:1746]: :+1:

**Review by wpaulino — APPROVED** [review]

File: discovery/gossiper.go [hunk 2]
```diff
@@ -1741,6 +1741,10 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 			} else {
 				log.Tracef("Router rejected channel "+
 					"edge: %v", err)
+
```
Comment (cfromknecht) [inline discovery/gossiper.go:1744]: nit: commit message needs wrapping

**Review by cfromknecht — APPROVED** [review]
LGTM 👍

**Pull request merged**